modules/apps/wiki/wiki-service/src/main/java/com/liferay/wiki/service/impl/WikiPageLocalServiceImpl.java

```diff
@@ -2281,10 +2297,33 @@ else if (status != WorkflowConstants.STATUS_IN_TRASH) {
 		return wikiPagePersistence.update(page);
 	}
 
+	@Override
+	public WikiPage updateWikiPage(WikiPage wikiPage) {
+		try {
+			_validateHierarchyCycle(
+				wikiPage.getNodeId(), wikiPage.getTitle(),
+				wikiPage.getParentTitle());
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+
+		return super.updateWikiPage(wikiPage);
+	}
+
 	@Override
 	public WikiPage updateWikiPage(
 		WikiPage wikiPage, ServiceContext serviceContext) {
 
+		try {
+			_validateHierarchyCycle(
+				wikiPage.getNodeId(), wikiPage.getTitle(),
+				wikiPage.getParentTitle());
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+
 		return wikiPagePersistence.update(wikiPage, serviceContext);
```

Comment:
Can you move these dangling pieces of code inside the try catch? I think it makes more explicit that the error/success branches are independent. Something like:
```

		try {
			_validateHierarchyCycle(
				wikiPage.getNodeId(), wikiPage.getTitle(),
				wikiPage.getParentTitle());

			return wikiPagePersistence.update(wikiPage, serviceContext);
		}
		catch (PortalException portalException) {
			throw new RuntimeException(portalException);
		}
```

Reply:
If we do so, a portal exception throw in wikiPagePersistence.update could be catched and converted in a Runtime exception.

To avoid it we could create a new type of exception (hierarchyCicleException) and use it in the try catch, What do you think?